app/controllers/discourse_subscriptions/subscribe_controller.rb

```diff
@@ -1,31 +1,55 @@
 # frozen_string_literal: true
 
 module DiscourseSubscriptions
-  class SubscriptionsController < ::ApplicationController
+  class SubscribeController < ::ApplicationController
     include DiscourseSubscriptions::Stripe
     include DiscourseSubscriptions::Group
     before_action :set_api_key
     requires_login
 
     def index
       begin
-        products = ::Stripe::Product.list(active: true)
+        product_ids = Product.all.pluck(:external_id)
+        products = []
+
+        if product_ids.present? && is_stripe_configured?
+          response = ::Stripe::Product.list({
+            ids: product_ids,
+            active: true
+          })
+
+          products = response[:data].map do |p|
+            serialize_product(p)
+          end
 
-        subscriptions = products[:data].map do |p|
-          {
-            id: p[:id],
-            description: p.dig(:metadata, :description)
-          }
         end
 
-        render_json_dump subscriptions
+        render_json_dump products
+
+      rescue ::Stripe::InvalidRequestError => e
+        render_json_error e.message
+      end
+    end
+
+    def show
+      begin
+        product = ::Stripe::Product.retrieve(params[:id])
```

Comment:
It wouldn't hurt to add `params.require(:id)` just before this line.

Reply:
Good catch!